The built-in qsort_r implementation detection/test thing is broken on
musl

https://github.com/rlancaste/stellarsolver/issues/125

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,29 +58,9 @@ if (WIN32)
 
 else(WIN32)
 
-try_run(RUN_RESULT_2 COMPILE_SUCCESS_2 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_DECLARE_QSORT_R=ON)
-if(COMPILE_SUCCESS_2 AND (RUN_RESULT_2 EQUAL 0))
-    SET(VAR_2 0)
-else()
-    SET(VAR_2 1)
-endif()
-file(APPEND "${config_FN}" "#define NEED_DECLARE_QSORT_R ${VAR_2}\n")
-
-try_run(RUN_RESULT_3 COMPILE_SUCCESS_3 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_QSORT_R=ON)
-if(COMPILE_SUCCESS_3 AND (RUN_RESULT_3 EQUAL 0))
-    SET(VAR_3 1)
-else()
-    SET(VAR_3 0)
-endif()
-file(APPEND "${config_FN}" "#define NEED_QSORT_R ${VAR_3}\n")
-
-try_run(RUN_RESULT_4 COMPILE_SUCCESS_4 ${CMAKE_CURRENT_BINARY_DIR} ${config_SRCS} COMPILE_DEFINITIONS -DTEST_SWAP_QSORT_R=ON)
-if(COMPILE_SUCCESS_4 AND (RUN_RESULT_4 EQUAL 0))
-    SET(VAR_4 1)
-else()
-    SET(VAR_4 0)
-endif()
-file(APPEND "${config_FN}" "#define NEED_SWAP_QSORT_R ${VAR_4}\n")
+file(APPEND "${config_FN}" "#define NEED_DECLARE_QSORT_R 0\n")
+file(APPEND "${config_FN}" "#define NEED_QSORT_R 0\n")
+file(APPEND "${config_FN}" "#define NEED_SWAP_QSORT_R 1\n")
 
 endif(WIN32)
 
--- a/stellarsolver/astrometry/include/astrometry/ioutils.h
+++ b/stellarsolver/astrometry/include/astrometry/ioutils.h
@@ -31,15 +31,6 @@ char* dirname(const char* path);
 
 extern uint32_t ENDIAN_DETECTOR;
 
-void QSORT_R(void* base, size_t nmembers, size_t member_size,
-             void* token, int (*compar)(void *, const void *, const void *));
-
-/**
-   You should define the "comparison" function like this:
-   static int QSORT_COMPARISON_FUNCTION(my_comparison, void* token, const void* v1, const void* v2) {
- */
-#define QSORT_COMPARISON_FUNCTION(func, thunk, v1, v2) func(thunk, v1, v2)
-
 int copy_file(const char* infn, const char* outfn);
 
 int pad_fid(FILE* fid, size_t len, char pad);
--- a/stellarsolver/astrometry/include/astrometry/permutedsort.h
+++ b/stellarsolver/astrometry/include/astrometry/permutedsort.h
@@ -7,7 +7,7 @@
 #define PERMUTED_SORT_H
 
 // for QSORT_COMPARISON_FUNCTION
-#include "ioutils.h"
+#include "os-features.h"
 
 /*
  Computes the permutation array that will cause the "realarray" to be
--- a/stellarsolver/astrometry/util/ioutils.c
+++ b/stellarsolver/astrometry/util/ioutils.c
@@ -43,7 +43,6 @@
 
 #include "os-features.h"
 #include "ioutils.h"
-#include "qsort_reentrant.c"
 #include "errors.h"
 #include "log.h"
 
--- a/tester/mainwindow.h
+++ b/tester/mainwindow.h
@@ -38,6 +38,7 @@
 #endif
 
 //Astrometry.net includes
+#include "astrometrylogger.h"
 extern "C"{
 #include "ioutils.h"
 #include "bl.h"
